frontend/components/SetupPage/Create/SetupCreateSafe.tsx

```diff
@@ -137,75 +111,49 @@ export const SetupCreateSafe = () => {
 
   const creationStatusText = useMemo(() => {
     if (isCreatingSafe) return 'Creating accounts';
-    if (masterSafeAddress) return 'Account created';
+    if (isSuccess) return 'Account created';
     return 'Account creation in progress';
-  }, [isCreatingSafe, masterSafeAddress]);
+  }, [isCreatingSafe, isSuccess]);
 
   useEffect(() => {
     if (
       /**
        * Avoid creating safes if any of the following conditions are met:
        */
-      [
-        // optimismFailed, baseFailed, ethereumFailed
-        gnosisFailed,
-      ].some((x) => x) || // any of the chains failed
-      isCreatingSafe //|| // already creating a safe
-      // [isBaseSuccess, isEthereumSuccess, isOptimismSuccess].some((x) => !x) // any of the chains are not successful
+      isFailed || // creation failed - it's retried in background
+      isCreatingSafe || // already creating a safe
+      !isWalletsFetched // wallets are not loaded yet
     )
       return;
 
-    const chainsToCreateSafesFor = {
-      // [MiddlewareChain.OPTIMISM]: masterSafeAddressKeyExistsForChain(
-      //   MiddlewareChain.OPTIMISM,
-      // ),
-      // [MiddlewareChain.ETHEREUM]: masterSafeAddressKeyExistsForChain(
-      //   MiddlewareChain.ETHEREUM,
-      // ),
-      // [MiddlewareChain.BASE]: masterSafeAddressKeyExistsForChain(
-      //   MiddlewareChain.BASE,
-      // ),
-      [MiddlewareChain.GNOSIS]: !!masterSafeAddress,
-    };
-
-    const safeCreationsRequired = Object.entries(chainsToCreateSafesFor).reduce(
-      (acc, [chain, safeAddressAlreadyExists]) => {
-        const middlewareChain = chain as MiddlewareChain;
-        if (safeAddressAlreadyExists) {
-          // switch (middlewareChain) {
-          //   case MiddlewareChain.OPTIMISM:
-          //     setIsOptimismSuccess(true);
-          //     break;
-          //   case MiddlewareChain.ETHEREUM:
-          //     setIsEthereumSuccess(true);
-          //     break;
-          //   case MiddlewareChain.BASE:
-          //     setIsBaseSuccess(true);
-          //     break;
-          // }
-
-          switch (middlewareChain) {
-            case MiddlewareChain.GNOSIS:
-              setIsGnosisSuccess(true);
-              break;
+    const chainsToCreateSafesFor = serviceTemplate
+      ? Object.keys(serviceTemplate.configurations)
+      : null;
+
+    const safeCreationsRequired = chainsToCreateSafesFor
+      ? chainsToCreateSafesFor.reduce((acc, chain) => {
+          const safeAddressAlreadyExists = masterSafes?.find(
+            (safe) => safe.evmChainId === asEvmChainId(chain),
+          );
+          if (!safeAddressAlreadyExists) {
+            const middlewareChain = chain as MiddlewareChain;
+            acc.push(middlewareChain);
           }
           return acc;
-        }
-        return [...acc, middlewareChain];
-      },
-      [] as MiddlewareChain[],
-    );
+        }, [] as MiddlewareChain[])
+      : [];
```

Comment:
no need to change for now immediately
should probably integrate this approach in AGENT_CONFIGS as `requiresMasterSafesOn: EvmChainId[];` is available via `selectedAgentConfig`